modulemd/v2/modulemd-module-index.c

```diff
@@ -201,10 +222,11 @@ add_subdoc (ModulemdModuleIndex *self,
 }
 
 
-static gboolean
+gboolean
```

Comment:
If we're making this a function that's callable from outside of this file (even if private), I think that it should have the `failures` initialization as well that are in the public functions.

Reply:
I avoided doing that because it would involve non-trivial changes to the other places it's called (and for little benefit). Everywhere else this function is called, it's appending to a failures list that may already have content. To change it would mean adding code to all the callers to iterate it and append its contents to another list. That's not impossible, but I'm not sure it's a valid use of time.

I documented in the header that @failures needed to be pre-initialized before calling this function.

Reply:
As discussed on IRC, I seem to have been vague in my comment.

I meant that you might just add what e.g. `update_from_stream` has:
````
if (*failures == NULL)                                 
  *failures = g_ptr_array_new_full (0, g_object_unref);
````

That should make it not change anything for other callers.

Reply:
OK, that seems reasonable. I'll do that.